CreateWorkflowInstanceTest is unstable

1. The test creates a new topic with three partitions
1. Verifies they are created by making repeated partitions requests to the system partition
1. Then uses the topology to obtain the partition ids

The problem is that in step three it may happen that not all partitions are in the topology response yet.

[stdout.txt](https://github.com/zeebe-io/zeebe/files/2201131/stdout.txt)

Stacktrace:

```
[ERROR] Tests run: 17, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 13.692 s <<< FAILURE! - in io.zeebe.broker.workflow.CreateWorkflowInstanceTest

[ERROR] shouldCreateWorkflowInstanceOnAllPartitions(io.zeebe.broker.workflow.CreateWorkflowInstanceTest)  Time elapsed: 1.152 s  <<< FAILURE!

java.lang.AssertionError: 


Expected size:<3> but was:<2> in:

<[4294967408L, 4294967408L]>

	at io.zeebe.broker.workflow.CreateWorkflowInstanceTest.shouldCreateWorkflowInstanceOnAllPartitions(CreateWorkflowInstanceTest.java:581)

	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)

	at java.lang.reflect.Method.invoke(Method.java:498)

	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)

	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)

	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)

	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)

	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)

	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)

	at org.junit.rules.RunRules.evaluate(RunRules.java:20)

	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)

	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)

	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

	at org.junit.runners.Suite.runChild(Suite.java:128)

	at org.junit.runners.Suite.runChild(Suite.java:27)

	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)

	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)

	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)

	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)

	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)

	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)

	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)

	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)

	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)

	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)

	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)

	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)

```

--- a/broker-core/src/test/java/io/zeebe/broker/workflow/CreateWorkflowInstanceTest.java
+++ b/broker-core/src/test/java/io/zeebe/broker/workflow/CreateWorkflowInstanceTest.java
@@ -560,7 +560,7 @@ public class CreateWorkflowInstanceTest {
     final int partitions = 3;
 
     apiRule.createTopic("test", partitions);
-    final List<Integer> partitionIds = apiRule.getPartitionsFromTopology("test");
+    final List<Integer> partitionIds = apiRule.getPartitionIds("test");
 
     final WorkflowDefinition definition =
         Bpmn.createExecutableWorkflow("process").startEvent().endEvent().done();

--- a/protocol-test-util/src/main/java/io/zeebe/test/broker/protocol/clientapi/ClientApiRule.java
+++ b/protocol-test-util/src/main/java/io/zeebe/test/broker/protocol/clientapi/ClientApiRule.java
@@ -83,13 +83,10 @@ public class ClientApiRule extends ExternalResource {
 
     msgPackHelper = new MsgPackHelper();
     streamAddress = transport.registerRemoteAddress(brokerAddress);
-    doRepeatedly(() -> getPartitionsFromTopology(Protocol.SYSTEM_TOPIC))
-        .until(Objects::nonNull, Objects::isNull);
 
     if (usesDefaultTopic) {
       final List<Integer> partitionIds =
-          doRepeatedly(() -> getPartitionsFromTopology(DEFAULT_TOPIC_NAME))
-              .until(p -> !p.isEmpty());
+          doRepeatedly(() -> getPartitionIds(DEFAULT_TOPIC_NAME)).until(p -> !p.isEmpty());
       defaultPartitionId = partitionIds.get(0);
     }
   }
@@ -298,41 +295,6 @@ public class ClientApiRule extends ExternalResource {
         .data()
         .done()
         .sendAndAwait();
-  }
-
-  @SuppressWarnings("unchecked")
-  public List<Integer> getPartitionsFromTopology(String topicName) {
-    final ControlMessageResponse response =
-        createControlMessageRequest()
-            .messageType(ControlMessageType.REQUEST_TOPOLOGY)
-            .data()
-            .done()
-            .sendAndAwait();
-
-    final Map<String, Object> topology = response.getData();
-    final List<Map<String, Object>> brokers = (List<Map<String, Object>>) topology.get("brokers");
-
-    final Set<Integer> partitionIds = new HashSet<>();
-    for (Map<String, Object> broker : brokers) {
-      final List<Map<String, Object>> brokerPartitionStates =
-          (List<Map<String, Object>>) broker.get("partitions");
-      for (Map<String, Object> brokerPartitionState : brokerPartitionStates) {
-        if (topicName.equals(brokerPartitionState.get("topicName"))) {
-          partitionIds.add((int) brokerPartitionState.get("partitionId"));
-        }
-      }
-    }
-    return new ArrayList<>(partitionIds);
-  }
-
-  public int getSinglePartitionId(String topicName) {
-    final List<Integer> partitionIds = getPartitionsFromTopology(topicName);
-    if (partitionIds.size() != 1) {
-      throw new RuntimeException(
-          "There are " + partitionIds.size() + " partitions of topic " + topicName);
-    } else {
-      return partitionIds.get(0);
-    }
   }
 
   public int getDefaultPartitionId() {

